delete.js

```diff
@@ -0,0 +1,38 @@
+import { success, failure } from "./libs/response-lib";
+import * as dynamoDbLib from "./libs/dynamodb-lib";
+import User from './model/user';
+import HttpStatus from 'http-status-codes';
+import { UUID_V1 } from './validators/user';
+
+/**
+ * Delete API Lambda for User model.
+ */
+export async function main(event, context) {
+  try {
+    if (!('pathParameters' in event)) {
+      return failure(HttpStatus.UNPROCESSABLE_ENTITY, {error : "ID Path Parameter required."});
+    }
+    const id = event.pathParameters.id;
+
+    if (!id) {
+      return failure(HttpStatus.UNPROCESSABLE_ENTITY, {error : "ID Path Parameter required."});
+    } else if (!id.match(UUID_V1)) {
+      throw { message : "Required UUID V1 format." };
+    }
+
+    let user = new User(dynamoDbLib);
+    try {
+      const exists = await user.read(id);
+      if (!exists) {
+        return failure(HttpStatus.NOT_FOUND, {});
```

Comment:
Wasteful in terms of?  I'm not sure I see how this impacts performance in a meaningful way, I understand an empty parameter is not really of any use, probably could set this as a default value then it wouldn't be required.

Reply:
It probably has none or negligible impact on performance. Just repeating this empty object over and over again instead of adding default to the argument seemed strange.